exercises/practice/react/test/react_test.exs

```diff
@@ -0,0 +1,264 @@
+defmodule ReactTest do
+  use ExUnit.Case
+
+  def send_callback(pid),
+    do: fn name, value -> send(pid, {:callback, name, value}) end
+
+  # @tag :pending
+  test "input cells have a value" do
+    {:ok, cells} = React.new([%{type: :input, name: "input", value: 10}])
+    assert React.get_value(cells, "input") == 10
+  end
+
+  @tag :pending
+  test "an input cell's value can be set" do
+    {:ok, cells} = React.new([%{type: :input, name: "input", value: 4}])
+
+    React.set_value(cells, "input", 20)
+    assert React.get_value(cells, "input") == 20
+  end
+
+  @tag :pending
+  test "compute cells calculate initial value" do
+    {:ok, cells} =
+      React.new([
+        %{type: :input, name: "input", value: 1},
+        %{type: :output, name: "output", inputs: ["input"], compute: fn a -> a + 1 end}
```

Comment:
I understand why you are using a map in your data model, to enforce a single entry per key, but I think when sending messages it is much more common to use a tuple for this.

```suggestion
        {:input, "input", 1},
        {:output, "output", ["input"], fn a -> a + 1 end}
```

Reply:
Actually, this is not why I use a tuple, here I use a map because I want to have named fields, otherwise it's unnecessarily complicated for the student to figure out which value is what... This is just following the data from `canonical-data.json`. 

If there was only two fields, it would be ok, but if I have sometimes 2, sometimes 3, I would always go for clarity. 
If you insist I will change to tuples, though.

Reply:
Well, maybe with the `@spec` specifying which is what it would be OK.

Reply:
I would rather we keep the maps precisely to have named fields. Tuples will be more confusing IMO, because there will be two different ones of different lengths... 

> but I think when sending messages 

This is not sending messages 🤔

Reply:
I mean "send a message" in a more philospohical sense ([archived article](https://web.archive.org/web/20170520194237/https://www.integralist.co.uk/posts/message-passing.html) since my link is dead).  Calling a function is "sending a message". Eg. when invoking the function `new` I am sending a message to create a system with these constraints.

Reply:
Tuples for this sort of thing is still super common, and we see tuples of varying lengths for various meanings all the time: `{:ok, value1, value2}` vs `{:error, "message"}`

Reply:
But that's why I suggested having a defined type for the tuple so that you can give meaning to the tuple

Reply:
How about we compromise: 
```elixir
%{type: :input, name: "input", value: 1},
{:output, "output", ["input"], fn a -> a + 1 end}
```
(🤣 I'm very sorry 😆)

I will reflect on it for a bit. Thank you both for the detailed review.

Reply:
Hmm, I see. Ok, I won't insist on maps, if @neenjaw is sure tuples will be better in this context, they will work too if we write a nice spec.